testpar/t_subfiling_vfd.c

```diff
@@ -2116,6 +2121,159 @@ parse_subfiling_env_vars(void)
     }
 }
 
+#ifdef H5_HAVE_FILTER_DEFLATE
+/*
+ * Test zlib.
+ */
+#define SUBF_FILENAME  "t_zlib.h5"
+#define SUBF_HDF5_TYPE H5T_NATIVE_INT
+#define SUBF_C_TYPE    int
+static void
+t_zlib(void)
+{
+    hsize_t start[1];
+    hsize_t count[1];
+    hsize_t dset_dims[1];
+    hsize_t chunk_dims[1];
+    size_t  target_size;
+    hid_t   file_id   = H5I_INVALID_HID;
+    hid_t   fapl_id   = H5I_INVALID_HID;
+    hid_t   dset_id   = H5I_INVALID_HID;
+    hid_t   dxpl_id   = H5I_INVALID_HID;
+    hid_t   fspace_id = H5I_INVALID_HID;
+    hid_t   plist_id  = H5I_INVALID_HID;
+    void   *buf       = NULL;
+
+    curr_nerrors = nerrors;
+
+    dxpl_id = H5Pcreate(H5P_DATASET_XFER);
+    VRFY((dxpl_id >= 0), "DXPL creation succeeded");
+    VRFY((H5Pset_dxpl_mpio(dxpl_id, H5FD_MPIO_COLLECTIVE) >= 0), "H5Pset_dxpl_mpio() succeeded");
+
+    /* Set selection I/O mode on DXPL */
+    VRFY((H5Pset_selection_io(dxpl_id, H5D_SELECTION_IO_MODE_ON) >= 0), "H5Pset_selection_io succeeded");
+
+    if (MAINPROCESS)
```

Comment:
Why isn't this first to be executed?

Reply:
Good catch - I don't know.
I just copied from the existing function.
@jhendersonHDF , do you know why?

Reply:
The above code was added by someone else. Probably best to move the MAINPROCESS checks up everywhere. I can do that after this is merged.